pkg/ccl/changefeedccl/changefeed_stmt.go

```diff
@@ -636,25 +642,61 @@ func createChangefeedJobRecord(
 	hasSelectPrivOnAllTables := true
 	hasChangefeedPrivOnAllTables := true
 	tolerances := opts.GetCanHandle()
-	for _, desc := range tableNameToDescriptor {
-		if table, isTable := desc.(catalog.TableDescriptor); isTable {
-			if err := changefeedvalidators.ValidateTable(targets, table, tolerances); err != nil {
-				return nil, changefeedbase.Targets{}, err
+	// If the user has database-level changefeed privileges, they can create
+	// table-level changefeeds on all tables in the database (except if they
+	// don't have select privileges on the table and it is a core changefeed)
+	if changefeedStmt.Level == tree.ChangefeedLevelTable {
+		_, tableToDatabaseLookup, err := buildTableToDatabaseAndSchemaLookup(tableAndParentDescs)
+		if err != nil {
+			return nil, changefeedbase.Targets{}, err
+		}
+		for _, desc := range tableNameToDescriptor {
+			if table, isTable := desc.(catalog.TableDescriptor); isTable {
+				if err := changefeedvalidators.ValidateTable(targets, table, tolerances); err != nil {
+					return nil, changefeedbase.Targets{}, err
+				}
+				for _, warning := range changefeedvalidators.WarningsForTable(table, tolerances) {
+					p.BufferClientNotice(ctx, pgnotice.Newf("%s", warning))
+				}
+
+				hasSelect, hasChangefeed, err := checkPrivilegesForDescriptor(ctx, p, desc)
+				if err != nil {
+					return nil, changefeedbase.Targets{}, err
+				}
+
+				databaseDesc, ok := tableToDatabaseLookup[table.GetID()]
+				if !ok {
+					return nil, changefeedbase.Targets{}, errors.Errorf("expected to find a database descriptor for table %s", table.GetName())
+				}
+				_, hasDatabaseChangefeedPriv, err := checkPrivilegesForDescriptor(ctx, p, databaseDesc)
```

Comment:
If we've already verified they have privileges on the table, we should avoid checking if they also have privileges on the db